worker/server.go

```diff
@@ -29,22 +29,25 @@ func (s Server) StartExtractingLinks(ctx context.Context, request *proto.StartEx
 	return &empty.Empty{}, nil
 }
 
-// Start grpc server for worker
+// Start will start a grpc server for worker
 func (s Server) Start(port int) error {
-	address := fmt.Sprintf(":%d", port)
-	lis, err := net.Listen("tcp", address)
+	gRPCService, err := service.
+		NewGRPCBuilder("Worker").
+		RegisterHandler(func(server *grpc.Server) {
+			proto.RegisterWorkerServer(server, &s)
+		}).
+		Build()
 	if err != nil {
-		return err
+		panic(err)
```

Comment:
We should return error ( handle it in the caller ) instead of panic ( crashing the server )

Reply:
We will handle the error at the caller level